core/src/main/java/com/linecorp/armeria/client/DefaultClientFactory.java

```diff
@@ -171,4 +173,19 @@ public void close() {
     void doClose() {
         clientFactoriesToClose.forEach(ClientFactory::close);
     }
+
+    private URI normalizeUri(URI uri, Scheme scheme) {
+        if (uri == HttpClientBuilder.UNDEFINED_URI) {
```

Comment:
Could you revert https://github.com/line/armeria/pull/2241/files#diff-8db0b28d88b37c5051e9775d4df020bfL41 and use `isUndefinedUri(URI uri)`? We want to hide `UNDEFINED_URI` instance from outside. 😉
```suggestion
        if (isUndefinedUri(uri)) {
```

Reply:
Done.